### ISSUE TITLE
PageCache is caching non-200 responses

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#22014]
If you build a template/page that sets a non-200 status for the response, e.g. `$response.sendRedirect("https://www.google.com)`, whatever was written in the response still gets cached by the page cache, even though a 301.

The first call to a page that is doing a redirect in velocity get the proper status, e.g. a `301` or a `302`, but subsequent calls to the same url will return a 200 response with the text of the page while that page is still in cache.

**Comment by jcastro-dotcms** [comment]
Passed Internal QA.

**NOTE TO QA:** Here are the steps I followed to verify the fix.

1. Create an HTML Page called `test-page`, with a TTL equal to `30`.
2. Add a Widget to the page with the Velocity code mentioned by Will:
```velocity
$response.sendRedirect("https://www.google.com")
```
3. Use the following CURL command to access the page:
```curl
curl -v http://localhost:8080/test-page
```
4. Here are the results **BEFORE** applying the fix -- notice how the second request returns a 200, which is not correct:
```
~/dotCMS/git/support/docker-images/single_node/release-22.06 ⌚ 15:51:18
$ curl -v http://localhost:8080/test-page
*   Trying ::1:8080...
* Connected to localhost (::1) port 8080 (#0)
> GET /test-page HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.77.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< x-dot-server: jcastro-dotcms-MacBook-Pro.local|4090eb993a
< Set-Cookie: JSESSIONID=F2FC0954421C953A2469249F7544CE2E; Path=/; Secure; HttpOnly
< Set-Cookie: opvc=51b0a67d-450e-4a45-93cf-71e7fd6de6da; Path=/; HttpOnly
< Set-Cookie: sitevisitscookie=1; Max-Age=153792000; Expires=Sun, 04-Apr-2027 21:51:24 GMT; Path=/; HttpOnly
< Set-Cookie: dmid=e2d84e10-767b-48ae-99a5-8d8e1be23465; Max-Age=153792000; Expires=Sun, 04-Apr-2027 21:51:25 GMT; Path=/; HttpOnly
< Cache-Control: max-age=30
< Location: https://www.google.com
< Connection: close
< Content-Type: text/html;charset=UTF-8
< Content-Length: 0
< Date: Fri, 20 May 2022 21:51:26 GMT
< 
* Closing connection 0

~/dotCMS/git/support/docker-images/single_node/release-22.06 ⌚ 15:51:26
$ curl -v http://localhost:8080/test-page
*   Trying ::1:8080...
* Connected to localhost (::1) port 8080 (#0)
> GET /test-page HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.77.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< x-dot-server: jcastro-dotcms-MacBook-Pro.local|4090eb993a
< Set-Cookie: JSESSIONID=887FDEAE4CEA776EFA6C9746576F4F9C; Path=/; Secure; HttpOnly
< Set-Cookie: opvc=ae9668e1-e48a-4ee1-966f-4e0d27d8c1ef; Path=/; HttpOnly
< Set-Cookie: sitevisitscookie=1; Max-Age=153792000; Expires=Sun, 04-Apr-2027 21:51:29 GMT; Path=/; HttpOnly
< Set-Cookie: dmid=03d7b5e7-305e-4c53-a4cd-3f4d9984797b; Max-Age=153792000; Expires=Sun, 04-Apr-2027 21:51:29 GMT; Path=/; HttpOnly
< Cache-Control: max-age=30
< Content-Type: text/html;charset=UTF-8
< Content-Length: 114
< Date: Fri, 20 May 2022 21:51:29 GMT
< 
<html>
    <header></header>
    <body>
        <h1>Hello World!</h1>
        
        
    </body>
* Connection #0 to host localhost left intact
</html>%                                                                 
```
5. Here are the results **AFTER** applying the fix -- notice how both requests returns a 302, which is correct:
```
~/dotCMS/git/support/docker-images/single_node/release-22.06 ⌚ 8:21:50
$ curl -v http://localhost:8080/test-page
*   Trying ::1:8080...
* Connected to localhost (::1) port 8080 (#0)
> GET /test-page HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.77.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< x-dot-server: jcastro-dotcms-MacBook-Pro.local|4090eb993a
< Set-Cookie: JSESSIONID=8AA447A10EBDE3EFEBB40BF67A6585C1; Path=/; Secure; HttpOnly
< Set-Cookie: opvc=8f70e31e-7c19-44ac-9ddd-3c7d1ae5c60a; Path=/; HttpOnly
< Set-Cookie: sitevisitscookie=1; Max-Age=153792000; Expires=Wed, 07-Apr-2027 14:21:55 GMT; Path=/; HttpOnly
< Set-Cookie: dmid=e9c4340f-d1b2-4891-9446-26554e46abfd; Max-Age=153792000; Expires=Wed, 07-Apr-2027 14:22:04 GMT; Path=/; HttpOnly
< Cache-Control: max-age=30
< Location: https://www.google.com
< Connection: close
< Content-Type: text/html;charset=UTF-8
< Content-Length: 0
< Date: Mon, 23 May 2022 14:22:05 GMT
< 
* Closing connection 0

~/dotCMS/git/support/docker-images/single_node/release-22.06 ⌚ 8:22:05
$ curl -v http://localhost:8080/test-page
*   Trying ::1:8080...
* Connected to localhost (::1) port 8080 (#0)
> GET /test-page HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.77.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< x-dot-server: jcastro-dotcms-MacBook-Pro.local|4090eb993a
< Set-Cookie: JSESSIONID=43DF815E38383E53D1CDC3B4831CB34B; Path=/; Secure; HttpOnly
< Set-Cookie: opvc=4082b5ea-5e86-4324-b2f3-aea06e15f9bd; Path=/; HttpOnly
< Set-Cookie: sitevisitscookie=1; Max-Age=153792000; Expires=Wed, 07-Apr-2027 14:22:10 GMT; Path=/; HttpOnly
< Set-Cookie: dmid=9db2f3a2-3fce-4c8b-a931-39fbb6be06d5; Max-Age=153792000; Expires=Wed, 07-Apr-2027 14:22:10 GMT; Path=/; HttpOnly
< Cache-Control: max-age=30
< Location: https://www.google.com
< Connection: close
< Content-Type: text/html;charset=UTF-8
< Content-Length: 0
< Date: Mon, 23 May 2022 14:22:10 GMT
< 
* Closing connection 0
```

**Comment by bryanboza** [comment]
Fixed, tested on release-22.06 // Docker // FF
